cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1508,51 +1528,118 @@ func (r machineResource) freeMachine(request *restful.Request, response *restful
 	err = response.WriteHeaderAndEntity(http.StatusOK, makeMachineResponse(m, r.ds, logger))
 	if err != nil {
 		logger.Error("Failed to send response", zap.Error(err))
-		return
 	}
 }
 
 func freeMachine(ds *datastore.RethinkStore, publisher bus.Publisher, ipamer ipam.IPAMer, m *metal.Machine, logger *zap.SugaredLogger) error {
-	if m.State.Value == metal.LockedState {
-		return fmt.Errorf("machine is locked")
-	}
-
-	// free machine should be callable even many times in a row (such that events can be fired multiple times if necessary)
-
-	_, err := setVrfAtSwitches(ds, m, "")
-	logger.Infow("set VRF at switch", "machineID", m.ID, "error", err)
+	err := deleteVRFSwitches(ds, m, logger)
 	if err != nil {
 		return err
 	}
-
-	deleteEvent := metal.MachineEvent{Type: metal.DELETE, OldMachineID: m.ID}
-	err = publisher.Publish(metal.TopicMachine.GetFQN(m.PartitionID), deleteEvent)
-	logger.Infow("published machine delete event", "machineID", m.ID, "error", err)
+	err = publishDeleteEvent(publisher, m, logger)
 	if err != nil {
 		return err
 	}
 
+	// if the machine is allocated, we free it in our database
 	if m.Allocation != nil {
-		// if the machine is allocated, we free it in our database
 		err = releaseMachineNetworks(ds, ipamer, m)
 		if err != nil {
 			// TODO: Trigger network garbage collection
 			// TODO: Check if all IPs in rethinkdb are in the IPAM and vice versa, cleanup if this is not the case
 			// TODO: Check if there are network prefixes in the IPAM that are not in any of our networks
-			logger.Errorw("an error during releasing machine networks occurred, scheduled network garbage collection", "error", err)
+			logger.Errorf("an error during releasing machine networks occurred, scheduled network garbage collection", "error", err)
+		}
+
+		old := *m
```

Comment:
Oops, you put that into the if condition, but this needs to be outside the condition... just as before.